homeassistant/components/shelly/entity.py

```diff
@@ -20,6 +20,31 @@ def temperature_unit(block_info: dict) -> str:
     return TEMP_CELSIUS
 
 
+def shelly_naming(self, block, entity_type: str):
+    """Naming for switch and sensors."""
+    channels = self.wrapper.device.shelly["num_outputs"]
```

Comment:
We have exception here for Shelly H&T:
```
Traceback (most recent call last):
  File "/home/maciek/develop/home-assistant-core/homeassistant/helpers/entity_platform.py", line 193, in _async_setup_platform
    await asyncio.shield(task)
  File "/home/maciek/develop/home-assistant-core/homeassistant/components/shelly/sensor.py", line 128, in async_setup_entry
    await async_setup_entry_attribute_entities(
  File "/home/maciek/develop/home-assistant-core/homeassistant/components/shelly/entity.py", line 73, in async_setup_entry_attribute_entities
    [
  File "/home/maciek/develop/home-assistant-core/homeassistant/components/shelly/entity.py", line 74, in <listcomp>
    sensor_class(wrapper, block, sensor_id, description, counts[sensor_id])
  File "/home/maciek/develop/home-assistant-core/homeassistant/components/shelly/entity.py", line 158, in __init__
    super().__init__(wrapper, block)
  File "/home/maciek/develop/home-assistant-core/homeassistant/components/shelly/entity.py", line 103, in __init__
    self._name = shelly_naming(self, block, "switch")
  File "/home/maciek/develop/home-assistant-core/homeassistant/components/shelly/entity.py", line 25, in shelly_naming
    channels = self.wrapper.device.shelly["num_outputs"]
KeyError: 'num_outputs'
```

Reply:
humm, thx for spotting. I suppose that mine was in sleep when I tested the code. Will recheck. I bet it needs a guard like for Shelly in roller mode.